src/main/java/seedu/address/model/appointment/AppointmentFilterByNricPredicate.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.appointment;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that an {@code Appointment}'s {@code Doctor}'s {@code Nric}
+ * and {@code Patient}'s {@code Nric} matches the predicates given.
+ */
+public class AppointmentFilterByNricPredicate implements Predicate<Appointment> {
+    private final Predicate<Appointment> patientPredicate;
+    private final Predicate<Appointment> doctorPredicate;
```

Comment:
would it be better to hold a list of predicates? so that more predicates can be added like start/end time